openedx/features/offline_mode/utils.py

```diff
@@ -18,85 +19,95 @@
-    """
-    try:
-        for resource_path in os.listdir(current_base_path):
-            full_path = os.path.join(current_base_path, resource_path)
-            full_path_in_zip = os.path.join(current_path_in_zip, resource_path)
-            if os.path.isfile(full_path):
-                zip_file.write(full_path, full_path_in_zip)
-            else:
-                add_files_to_zip_recursively(zip_file, full_path, full_path_in_zip)
-    except OSError:
-        log.error(f'Error while reading the directory: {current_base_path}')
-        return
+    def __init__(self, storage_class=None, storage_kwargs=None):
+        if storage_kwargs is None:
+            storage_kwargs = {}
+
+        self.storage = get_storage(storage_class, **storage_kwargs)
+
+    def generate_offline_content(self, xblock, html_data):
+        """
+        Generates archive with XBlock content for offline mode.
+
+        Args:
+            xblock (XBlock): The XBlock instance
+            html_data (str): The rendered HTML representation of the XBlock
+        """
+        if not is_modified(xblock):
+            return
+
+        base_path = block_storage_path(xblock)
+        remove_old_files(xblock)
+        tmp_dir = mkdtemp()
+
+        try:
+            self.save_xblock_html(tmp_dir, xblock, html_data)
+            self.create_zip_file(tmp_dir, base_path, f'{xblock.location.block_id}.zip')
+        finally:
+            shutil.rmtree(tmp_dir, ignore_errors=True)
+
+    @staticmethod
+    def save_xblock_html(tmp_dir, xblock, html_data):
+        """
+        Saves the XBlock HTML content to a file.
+
+        Generates the 'index.html' file with the HTML added to use it locally.
+
+        Args:
+            tmp_dir (str): The temporary directory path to save the xblock content
+            xblock (XBlock): The XBlock instance
+            html_data (str): The rendered HTML representation of the XBlock
+        """
+        html_manipulator = HtmlManipulator(xblock, html_data, tmp_dir)
+        updated_html = html_manipulator.process_html()
+
+        with open(os.path.join(tmp_dir, 'index.html'), 'w') as file:
+            file.write(updated_html)
+
+    def create_zip_file(self, temp_dir, base_path, file_name):
+        """
+        Creates a zip file with the Offline Content in the media storage.
+
+        Args:
+            temp_dir (str): The temporary directory path where the content is stored
+            base_path (str): The base path directory to save the zip file
+            file_name (str): The name of the zip file
+        """
+        with ZipFile(temp_dir + '/' + file_name, 'w') as zip_file:
+            zip_file.write(os.path.join(temp_dir, 'index.html'), 'index.html')
+            self.add_files_to_zip_recursively(
+                zip_file,
+                current_base_path=os.path.join(temp_dir, 'assets'),
+                current_path_in_zip='assets',
+            )
+        with open(temp_dir + '/' + file_name, 'rb') as buffered_zip:
```

Comment:
Here the same

Reply:
This filepath can be moved to variable, to be used here and above.

Reply:
Yes, made this change and above